src/common/entity/domain_icon.ts

```diff
@@ -71,13 +35,122 @@ export const domainIcon = (domain: string, state?: string): string => {
           return "hass:bell";
       }
 
-    case "binary_sensor":
-      return state && state === "off"
-        ? "hass:radiobox-blank"
-        : "hass:checkbox-marked-circle";
+    case "binary_sensor": {
+      const is_off = compareState === "off";
+      switch (stateObj?.attributes.device_class) {
+        case "battery":
+          return is_off ? "hass:battery" : "hass:battery-outline";
+        case "battery_charging":
+          return is_off ? "hass:battery" : "hass:battery-charging";
+        case "cold":
+          return is_off ? "hass:thermometer" : "hass:snowflake";
+        case "connectivity":
+          return is_off ? "hass:server-network-off" : "hass:server-network";
+        case "door":
+          return is_off ? "hass:door-closed" : "hass:door-open";
+        case "garage_door":
+          return is_off ? "hass:garage" : "hass:garage-open";
+        case "gas":
+        case "power":
+        case "problem":
+        case "safety":
+        case "smoke":
+          return is_off ? "hass:shield-check" : "hass:alert";
+        case "heat":
+          return is_off ? "hass:thermometer" : "hass:fire";
+        case "light":
+          return is_off ? "hass:brightness-5" : "hass:brightness-7";
+        case "lock":
+          return is_off ? "hass:lock" : "hass:lock-open";
+        case "moisture":
+          return is_off ? "hass:water-off" : "hass:water";
+        case "motion":
+          return is_off ? "hass:walk" : "hass:run";
+        case "occupancy":
+          return is_off ? "hass:home-outline" : "hass:home";
+        case "opening":
+          return is_off ? "hass:square" : "hass:square-outline";
+        case "plug":
+          return is_off ? "hass:power-plug-off" : "hass:power-plug";
+        case "presence":
+          return is_off ? "hass:home-outline" : "hass:home";
+        case "sound":
+          return is_off ? "hass:music-note-off" : "hass:music-note";
+        case "vibration":
+          return is_off ? "hass:crop-portrait" : "hass:vibrate";
+        case "window":
+          return is_off ? "hass:window-closed" : "hass:window-open";
+        default:
+          return is_off ? "hass:radiobox-blank" : "hass:checkbox-marked-circle";
+      }
+    }
+
+    case "cover": {
+      const open = compareState !== "closed";
```

Comment:
Why did you move all domain-specific icon code in here instead of importing the function?

Reply:
We need to default after wards. If the conditions aren't met we run the last lines of the domain icon

Reply:
Wouldnt this on top be enough? If the domain has a specific icon function let that handle it...
```
  if (domain in domainIcons) {
    return domainIcons[domain](state);	
  }
```

Reply:
Some domains have a specific icon but also non specific. Those are a fallback